### ISSUE TITLE
Extend parser to support subdot and branch syntax with new methods

### ISSUE THREAD

**Issue by lukemarsden** [dotmesh-io/python-sdk#4]
https://github.com/dotmesh-io/python-sdk/commit/9016db0359e6a3497529abd9ce7905973a82ed09 introduced a parser which will support splitting out namespace and name from a dot name. Per the original description of https://github.com/dotmesh-io/python-sdk/issues/3, extend the parser to support subdot and branch syntax, and make it fail when subdot and branch syntaxes are passed to the existing methods.

**Comment by mhausenblas** [comment]
So, time to cut a new release?

**Comment by lukemarsden** [comment]
Yes please, I already bumped the version but lack a PyPI account :'(

Although I prefer my phone's autocorrect of "I lack a puppy"

**Comment by mhausenblas** [comment]
He he, lacking a puppy is sad but maybe I can help … so I wanted to add you to the [collaborators of that package](https://pypi.org/manage/project/datadots-api/collaboration/) so that I'm not a bottleneck here, if you let me know which handle to use?
